.github/workflows/precommit-publish.yml

```diff
@@ -2,7 +2,8 @@ name: Create and publish Precommit Image
 
 on:
   push:
-    branches: [ "main" ]
+    paths:
```

Comment:
this seems to be based on the doc [here](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpushpull_requestpull_request_targetpathspaths-ignore)